Lib/enum.py

```diff
@@ -801,24 +801,12 @@ def __call__(cls, value, names=None, *, module=None, qualname=None, type=None, s
 
     def __contains__(cls, member):
         """
-        Return True if member is a member of this enum
-        raises TypeError if member is not an enum member
-
-        note: in 3.12 TypeError will no longer be raised, and True will also be
-        returned if member is the value of a member in this enum
+        Return True if member is a member of this enum or if member is the
+        value of a member in this enum.
         """
         if not isinstance(member, Enum):
-            import warnings
-            warnings.warn(
-                    "in 3.12 __contains__ will no longer raise TypeError, but will return True or\n"
-                    "False depending on whether the value is a member or the value of a member",
-                    DeprecationWarning,
-                    stacklevel=2,
-                    )
-            raise TypeError(
-                "unsupported operand type(s) for 'in': '%s' and '%s'" % (
-                    type(member).__qualname__, cls.__class__.__qualname__))
-        return isinstance(member, cls) and member._name_ in cls._member_map_
+            return member in cls._value2member_map_ or member in cls._unhashable_values_
+        return member._name_ in cls._member_map_
```

Comment:
just curious - with the isinstance(member, cls) gone, does that mean it's now true that Foo.X in Bar <=> Bar.X in Foo? I.e. members of the same name across enums? Name-based aliasing like this doesn't seem intentional. Maybe a test case should cover this, either way

Reply:
Very good point.  It should probably be:

        if isinstance(member, cls):
            return True
        return member in cls._value2member_map_ or member in cls._unhashable_values_

If they are both `IntEnums` then `Foo.X in Bar` <=> `Bar.X in Foo`  would be `True` only if both `X`s had the same value, not because they had the same name.

This definitely needs a test.